src/worldclim/futures.jl

```diff
@@ -0,0 +1,60 @@
+#http://biogeo.ucdavis.edu/data/worldclim/v2.1/fut/10m/wc2.1_10m_bioc_BCC-CSM2-MR_ssp126_2021-2040.zip
+#http://biogeo.ucdavis.edu/data/worldclim/v2.1/fut/10m/wc2.1_10m_tmin_BCC-CSM2-MR_ssp126_2021-2040.zip
+
+"""
+    getraster(T::Type{WorldClim{BioClim}}, [layer::Integer]; res::String="10m") => String
+    getraster(T::Type{WorldClim{BioClim}}, layer::Integer, res::String)
+
+Download WorldClim weather data, choosing `layer` from `$(layers(WorldClim{BioClim}))`,
+and `res` from `$(resolutions(WorldClim{BioClim}))`.
+
+Without a layer argument, all layers will be downloaded, and a tuple of paths is returned. 
+If the data is already downloaded the path will be returned.
+"""
+function getraster(T::Type{WorldClim{BioClim}}, ::Type{F}, layer::Integer, date=Year(2050); res::String=defres(T)) where {F <: FutureClimate}
```

Comment:
Again probably better if this is a single type argument to keep things standardised

Reply:
It's going to be a big type though - do you have an idea of what the type would look like? I ended up picking this syntax because it makes sense to me from a semantic point of view (we want the data, and then we want them in the future).

Reply:
Yeah. Its not the best, but I think method signature consistency is overall more important.

So:
```julia
getraster(T::Type{WorldClim{Future{BioClim}}}, layer::Integer, date=Year(2050); res::String=defres(T))
```

Reply:
This also makes writing the GeoData.jl wrapper easier, it's one function for future bioclim and bioclim, instead of 2.

Reply:
So I have a few counter-arguments for that.

1. I think it makes sense to separate two distinct elements, i.e. the type of data we want (bioclim from CHELSA) to the specific (no argument: extant data, argument: future data). 
2. I also think it makes sense because we can define the "future" part independently from the data source - multiple data sources will have future scenarios, and we want to be able to specify them correctly
3. the type signature you give as an example isn't actually shorter in practice because there is not a valid default model for the future climate, so the user still needs to specify all of it when getting the data
4. I get that this makes the GeoData wrapper simpler to write, but it's not going to be a majority of datasets at the moment
5. I think having the data source contains sometimes a dataset and sometimes a future data specification is actually less consistent for the user - I agree that the date should be kept outside of it, tho

If you think of it in terms of path, as well, I think it would make sense that `CHELSA{BioClim}, Future{X,Y}` goes into `CHELSA/BioClim/Future/X_Y...` - there is an implicit semantics to the types that should also be present in the folder hierarchy.

Almost all functions of this type in Julia accept multiple inputs, I think that as long as it is properly documented, it would make things _less_ confusing for users. An example is if you want to change from extant to future data: with a second type argument, you simply need to _add it_ to the function call. With nested types, you need to go into the type and edit it. I'd rather pay the cost of writing more methods for the datasets who require it than have an akward typing, especially since we have multiple dispatch for this reason.

Reply:
I here where you are coming from here, so maybe I'll explain myself a little more.A lot of the work I did to this package was to introduce strict stardardiations. Partly for users, but mostly for devs and for interop and reducing workload downstream. Not just in GeoData.jl but every package and script that builds on this in future. This is the only way I can maintain 20 or so julia packages - simplicity and strict standardisations. It also helps everyone else building on this in future.

This PR means now instead of one consistent method signature, this package has 2. I'm_really_ strongly against that happening for only aesthetic reasons. There is no technical reason 3 args is better than 2 here.

1. I think it makes sense to separate two distinct elements, i.e. the type of data we want (bioclim from CHELSA) to the specific (no argument: extant data, argument: future data). 

This could be argued either way.

2. I also think it makes sense because we can define the "future" part independently from the data source - multiple data sources will have future scenarios, and we want to be able to specify them correctly

This is possible with `Future` how I defined it? That generality was the point of removing the `...Climate` part... You can add as many params for it as you want, and use however many you need to in each case?  The trailing params are just ignored.

3. the type signature you give as an example isn't actually shorter in practice because there is not a valid default model for the future climate, so the user still needs to specify all of it when getting the data

Its still shorter, because future is just `Future`, not `FutureClimate`, and you have an extra space and comma. So 2 chars at minimum ;)

4. I get that this makes the GeoData wrapper simpler to write, but it's not going to be a majority of datasets at the moment

Not just geodata, ever wrapper anyone wants to write. Now there are 2 methods instead of 1. Downstream packages and scripts shouldn't know or care if its bioclim or future bioclim be passed through. We need to consider ecosystem consequences.

5. I think having the data source contains sometimes a dataset and sometimes a future data specification is actually less consistent for the user - I agree that the date should be kept outside of it, tho

I dont understand this point... but again its an aesthetic choice not a technical one. The user consistency I mean is argument number and placement, which this complicates.

How I organised this package was that besides source and layer all other args should be exposed as kw arguments, so as not to require more methods everywhere downstream. It can still be improved, but I think its fairly simple currently. But I think we should push to standardise it _even more_.

I can edit this into the PR, im just mostly here from my phone and busy currently.